### DISCUSSION TITLE
How to get base directory of hexo from plugin

### DISCUSSION THREAD

**Discussion by dimaslanjaka** [hexojs/hexo#5181] [Question / Answer] [answered]
Basically base directory is `process.cwd()` when called by CLI. but, how determine the base directory when hexo called by api `new Hexo` ?

**Comment by stevenjoezhang** [comment, marked as answer]
Use `hexo.base_dir`:
https://github.com/hexojs/hexo/blob/7edbf25d5b4c4d7fdf99ff2439cfd9b2cb1cfe0d/lib/hexo/index.js#L100

**Reply by dimaslanjaka** [reply]
so, it can be called like `hexo.base_dir`?

example:
```typescript
function getTempFolder(hexo: import('hexo')) {
  return hexo.base_dir + '/tmp';
}
```

thats right?

**Reply by stevenjoezhang** [reply]
Yes, `base_dir` is a property of global `hexo` object.
